File: Algorithm-CPP/17_선생님퀴즈.cpp
#include <iostream>
using namespace std;

int add(int n) {
    int sum = 0;

    for(int i = 1; i <= n; i++) {
        sum += i;
    }

    return sum;
}

int main() {
    int n;
    cin >> n;

    for(int i = 0; i < n; i++) {
        int num, answer;
        cin >> num >> answer;

        if (add(num) == answer) {
            cout << "YES" << endl;
        } else {
            cout << "NO" << endl;
        }
    }

    return 0;
}

File: Algorithm-Swift/Algorithm-Swift/Inflearn/17_선생님퀴즈.swift
//
//  17_선생님퀴즈.swift
//  Algorithm-Swift
//
//  Created by SHIN YOON AH on 2023/07/31.
//

import Foundation

func add(_ num: Int) -> Int {
    return (1...num).reduce(0, +)
}

func solution17() {
    let n = Int(readLine()!)!

    for _ in 0..<n {
        let nums = readLine()!.split(separator: " ").map { Int($0)! }
        let num = nums.first!
        let answer = nums.last!

        if answer == add(num) {
            print("YES")
        } else {
            print("NO")
        }
    }
}
